src/header/common/cookie.rs

```diff
@@ -27,10 +28,15 @@ use std::str::from_utf8;
 ///    ])
 /// );
 /// ```
-#[derive(Clone, PartialEq, Debug)]
-pub struct Cookie(pub Vec<String>);
+#[allow(missing_debug_implementations)]
+#[derive(Clone)]
+pub struct Cookie {
+    cookies: Vec<(String, String)>,
+    index: VecMap<String, String>,
```

Comment:
The `VecMap` is sufficient on it's own. There no need to double the strings here. The `VecMap` has `insert` and `append`, and `iter` gets all the key-value pairs, with duplicates.

Also, a `Cow<'static, str>` can be used, which allows someone who wants to set the same key and value on all cookies to do so without allocating and copying memory for it:

```rust
let mut cookie = Cookie::new();
// no allocation or copy required
cookie.set("foo", "bar");
// dynamic string, use a String
cookie.set(nonce_key.to_string(), nonce_val.to_string());
```